python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java

```diff
@@ -141,8 +145,44 @@ private FunctionType buildFunctionType(FunctionDef functionDef) {
 
   @Override
   public void visitImportName(ImportName importName) {
-    //createImportedNames(importName.modules(), null, Collections.emptyList());
-    super.visitImportName(importName);
+    importName.modules()
+      .forEach(aliasedName -> {
+        var names = aliasedName.dottedName().names();
+        var fqn = names
+              .stream().map(Name::name)
+              .toList();
+        var module = projectLevelTypeTable.getModule(fqn);
+        setTypeToName(aliasedName.alias(), module);
+
+        for (int i = names.size() - 1; i >= 0; i--) {
+          setTypeToName(names.get(i), module);
+          module = Optional.ofNullable(module)
+            .map(ModuleType::parent)
+            .orElse(null);
+        }
+      });
+  }
+
+  @Override
+  public void visitImportFrom(ImportFrom importFrom) {
```

Comment:
We should probably have a ticket about wildcard imports and make sure we don't have issues with them in the current implementation.

Reply:
agreed, [SONARPY-1781](https://sonarsource.atlassian.net/browse/SONARPY-1781)